src/nodes.coffee

```diff
@@ -2233,6 +2244,42 @@ exports.Parens = class Parens extends Base
 # string concatenation inside.
 
 exports.StringWithInterpolations = class StringWithInterpolations extends Parens
+  unwrap: -> this
```

Comment:
This is a fix for a bug that appears when we compile all interpolated strings to template literals (i.e. CS2). The problem didn't show up in any tests, and I haven't successfully written a test that captures the issue.

To see the bug, write a program that just contains `"#{2}"`. Without `unwrap`, the output is the old `"" + 2;`, and not the new and expected \`2\`. I tracked the problem down to a [line in Block.compileNode](https://github.com/jashkenas/coffeescript/blob/master/src/nodes.coffee#L290), which calls `unwrapAll`, before calling `compileNode` on the result. `unwrapAll` calls `unwrap` on each descendant child node until the result doesn't change, so it seems to strip away unnecessary(?) nodes in the hierarchy, before generating code. So the bug is caused by `unwrap` on `StringWithInterpolations`, which originally returned the `body` node inside it. Block.compileNode then called compileNode on this body node (mostly an "Op + node), which returns `"" + 2;`. My investigations show that the body node can also be an "Value>StringLiteral" node in the case of empty interpolated strings. 

So it seems that before this PR, code for an interpolated string can actually be generated in two ways: by StringWithInterpolations.compileNode, or by something reaching through to the body node within it (using `unwrap`), and calling compileNode on that. By changing `unwrap` to return `this`, and not the body node, I believe the second case should not happen: all code should be generated by StringWithInterpolations.compileNode. Since StringWithInterpolations.compileNode now holds the custom logic for outputting interpolated strings as template literals, it should be the only thing responsible for code generation.

In summary:
* I believe the new `unwrap` is definitely required in CS2 when all interpolated strings become template literals. If I'm right about what `unwrap` does, then StringWithInterpolations.compileNode should be the only piece of code responsible for JS-code generation, and never a child.
* I don't believe `unwrap` is needed for CS1 tagged template literals, as I don't believe Block.compileNode is triggered: the AST always contains a function-call node, with a interpolated node as a child node. But I could be wrong!
* I would be more comfortable if there was a test for this, and someone with more CS compiler knowledge than me had a opinion about this change. I made the change, it fixed the problem, all tests still pass, so I rolled with it :-)